### ISSUE TITLE
[BUG] SAR normalization leads to true_divide runtime warning

### ISSUE THREAD

**Issue by gramhagen** [recommenders-team/recommenders#858]
### Description
Getting a division by zero warning when using SAR normalize on movielens dataset.

### In which platform does it happen?
Using staging version of SAR on Windows.

### How do we replicate the issue?
 repeat issue with following code:
```
from reco_utils.recommender.sar import SAR
from reco_utils.dataset.movielens import load_pandas_df
from reco_utils.dataset.python_splitters import python_stratified_split

train, test = python_stratified_split(load_pandas_df())
model = SAR(normalize=True)
model.fit(train)
result = model.score(test, normalize=True)
```

generates the following warning
```
C:\Users\scgraham\repos\Recommenders\reco_utils\recommender\sar\sar_singlenode.py:316: RuntimeWarning: invalid value encountered in true_divide
  self.unity_user_affinity.dot(self.item_similarity)[user_ids, :]
```

the result.shape[0] = 943 whereas test.shape[0] = 25008
so it looks like it is aborting calculation for users when the warning occurs.

### Expected behavior (i.e. solution)
Avoid / handle divide by zero appropriately

**Comment by anargyri** [comment]
@gramhagen This warning should appear when dividing 0 by 0 (and not if you divide nonzero by 0). Therefore, probably what happens is that for some user u, all the rated items have similarity equal to 0 with a specific item i. E.g. i has no co-occurrence with the user's items.
I propose we set the score to -infinity in such cases, since user u is not connected in any way to item i in the training data. In other words, users who have watched the same movies as u have not watched movie i. This is what you already do in line 319, so no change is needed.

**Comment by miguelgfierro** [comment]
Closing old issue, feel free to reopen if needed.
